src/spec_tools/json_schema.cljc

```diff
@@ -202,7 +202,10 @@
 (defmethod accept-spec 'clojure.spec/and [dispatch spec children]
   (simplify-all-of {:allOf children}))
 
-; merge
+(defmethod accept-spec 'clojure.spec/merge [dispatch spec children]
+  {:type "object"
+   :properties (apply merge (map :properties children))
+   :required (apply concat (map :required children))})
```

Comment:
I'm not sure. You should try what Spec says when trying to conform a map with such spec.